buildSrc/src/main/kotlin/kotest-publishing-conventions.gradle.kts

```
@@ -103,9 +124,11 @@ val mavenPublishLimiter =
       maxParallelUsages = 1
    }
 
-tasks.withType<PublishToMavenRepository>().configureEach {
-   usesService(mavenPublishLimiter)
-}
+tasks.withType<PublishToMavenRepository>()
+   .matching { it.name.endsWith("PublicationTo${mavenCentralRepoName}Repository") }
+   .configureEach {
+      usesService(mavenPublishLimiter)
+   }
```

Isn't `tasks.withType` an eager API? If so:

```suggestion
tasks
   .named { it.name.endsWith("PublicationTo${mavenCentralRepoName}Repository") }
   .configureEach {
      usesService(mavenPublishLimiter)
   }
```

---

iirc `tasks.withType<> { ... }` is eager, but `tasks.withType<>().configureEach { ... }` is lazy.

---

Maybe that's the case. Their API surface is, well, amazingly complex. [Docs say](https://docs.gradle.org/current/userguide/task_configuration_avoidance.html)
> [TaskCollection.withType()](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/TaskCollection.html#withType-java.lang.Class-)
Ok to use. If chained withType().getByName(), use TaskCollection.named() instead.

If that's the case, the above would just be shorter as the task subtype is not required for `usesService()`.

---

And from what I understand the new `named {}` util is bugged and can sometimes be eager, so I try to avoid it.

https://github.com/gradle/gradle/issues/28347

---

Makes sense, excellent!

---

yes, that's true, `withType<>()` isn't needed here. But I think it makes it clearer that the code should only be configuring publishing tasks. Also, I'm not a big fan of string-based checking, so I feel a bit more comfortable with a type check :)

---

Yeah, I can relate. Although here, string matching is the decisive part, and the extra type check is a double-edged sword, providing an extra guard as well as an extra failure mode.